OpenRA.Mods.Common/Traits/Turreted.cs

```diff
@@ -102,19 +103,24 @@ public virtual void Tick(Actor self)
 			// NOTE: FaceTarget is called in AttackTurreted.CanAttack if the turret has a target.
 			if (attack != null)
 			{
-				if (!attack.IsAttacking)
+				if (!attack.IsAttacking || attack.IsTraitDisabled)
 				{
-					if (realignTick < info.RealignDelay)
-						realignTick++;
-					else if (info.RealignDelay > -1)
-						DesiredFacing = null;
-
-					MoveTurret();
+					attack = null;
 				}
 			}
 			else
```

Comment:
Can't this be else if (DesiredFacing != null) or do we need to call MoveTurret with the DesiredFacing as null?